lib/checkother.cpp

```diff
@@ -1408,34 +1408,45 @@ void CheckOther::checkConstVariable()
             if (usedInAssignment)
                 continue;
         }
-        // Skip if we ever cast this variable to a non-const type
+        // Skip if we ever cast this variable to a pointer/reference to a non-const type
         {
             bool castToNonConst = [&]
                 {
                     for (const Token* tok = var->nameToken(); tok != scope->bodyEnd && tok != nullptr; tok = tok->next())
                     {
                         if (tok->isCast())
                         {
+                            const Token* typeStart = nullptr; //first token of the type we're casting to
```

Comment:
I think this `typeStart` and `typeEnd` is an interesting approach .. it could be an acceptable first step until we have improved `valueType`.

Reply:
I actually wrote this before I started trying to fix typeValue. It's just pushed here so I don't lose it but it will hopefully be unnecessary if I use typeValue. It's actually not as effective as I thought it was since figuring out the complexities of the type token sequence is far from trivial even in this fairly simple case.